### DISCUSSION TITLE
Downtime for https://search.kokomo.cloud

### DISCUSSION THREAD

**Discussion by kokomo123** [searxng/searx-instances#377] [Report instance downtime]
Hi there, it seems my instance on Oracle keeps going down for some reason and this will affect SearXNG and bots were attacking the server, so I might need to take it down and try to harden it / reinstall it. If this doesn't help the issue, I may remove it from the list. Thank you for understanding.

**Comment by unixfox** [comment]
Thank you for the notification.

I wish you to be able to block bots.

**Reply by kokomo123** [reply]
Thank you for your well wishes. I managed to hopefully get crowdsec working and block any bad bots. I'll be monitoring searxng. and hope it wont crash or the vps doesnt crash. I'll mark this conversation as resolved for now and will reopen if the need arises, so i avoid duplicate discussions.

**Reply by unixfox** [reply]
If you want you can try the new anti bot built-in in searxng: https://github.com/searxng/searxng/blob/master/searx/botdetection/limiter.toml

It's still in testing phase that's why it's not announced yet. But it should help by a lot in combating bots.

**Reply by kokomo123** [reply]
is the limiter: true option all i need to turn it on?

**Reply by unixfox** [reply]
No it's an entirely separate setting, that's the one I've sent above.
You need to enable `link_token: true`

**Reply by kokomo123** [reply]
Ah, thank you. 👍🏻

**Reply by kokomo123** [reply]
I believe i enabled it by putting a limiter.toml in /etc/searxng and setting link_token: true, so fingers crossed it'll limit bots.

**Reply by return42** [reply]
> so fingers crossed it'll limit bots.

@kokomo123 the limiter and `link_token` method are described here: https://docs.searxng.org/src/searx.botdetection.html

The limiter and its additional methods can block bot requests to the SearXNG instance (Application Layer Security), they are not a general protection of a WEB server.

But to me it seems you have issue with the setup of your subdomain:

- https://kokomo.cloud/ --> sends a HTTP response
- https://search.kokomo.cloud --> ends with a timeout / no response

**Reply by kokomo123** [reply]
kokomo.cloud isn't meant to show anything right now as it's just a page to delegate another service unrelated to searxng. when search.kokomo.cloud times out, its showing the instance is down. I have to figure out why its OOM or bot attacking.

**Reply by kokomo123** [reply]
As a side note, link_token is enabled at the moment. Crowdsec might be ooming my instance so i'm gonna monitor this setup without crowdsec and just the limiter for now.
